xbmc/filesystem/AudioBookFileDirectory.cpp

```diff
@@ -81,29 +171,101 @@ bool CAudioBookFileDirectory::GetDirectory(const CURL& url,
     std::string chaptitle = StringUtils::Format(g_localizeStrings.Get(25010), i + 1);
     std::string chapauthor;
     std::string chapalbum;
+
+    std::shared_ptr<CFileItem> item(new CFileItem(url.Get(), false));
+    *item->GetMusicInfoTag() = albumtag;
+
+    auto addRole = [&](const std::string& role, const std::string& value)
+    {
+      if (!value.empty())
+        item->GetMusicInfoTag()->AddArtistRole(role, StringUtils::Split(value, separators));
```

Comment:
same as genre, maybe a more generic list of separators would be more appropriate here.
probably applies to involved people and a few other places as well.

Reply:
> maybe a more generic list of separators would be more appropriate here.
Like what?  This uses the internally defined separators, plus the user-defined separator in AS.xml.